### DISCUSSION TITLE
Moving folders internally on Sharepoint is not detected by onedrive

### DISCUSSION THREAD

**Discussion by esbenph** [abraunegg/onedrive#2303] [General]
The onedrive system by abraunegg usually works wonderfully for me. But today I discovered that after I have moved (i.e. archived) sub-folders from my Documents folder to other places on my OneDrive these changes are not executed in my local onedrive folders on Ubuntu.  The folders that I have moved away from Documents (in the web application Sharepoint) are not removed locally when I syncronize via the onedrive app.

Is this an error, or am I doing something in the wrong way?

**Comment by abraunegg** [comment]
@esbenph 
Firstly, what application version are you using? What is the output of `onedrive --version` ? 

> But today I discovered that after I have moved (i.e. archived) sub-folders from my Documents folder to other places on my OneDrive these changes are not executed in my local onedrive folders on Ubuntu.

Where did you 'move' them to? Another folder on your main 'account' or to a different SharePoint site? Is the place you 'moved' the files to 'excluded' by any client side filtering rules such as 'skip_dir' or 'sync_list' ?

**Comment by esbenph** [comment]
The version is onedrive v2.4.23-1+np1

Under my OneDrive account I moved some subfolders of the Documents folder on my OneDrive account to another folder on my OneDrive account on OneDrive. That is:
/'main-account'/Documents/subfolder1 was moved to
/'main-account'/Archive-folder/

(likewise with other subfolders)

When I then use the onedrive app on Ubuntu I run the following in Terminal:

onedrive --enable-logging --log-dir /home/esben/onedrivelog/ --single-directory "Documents" --synchronize

And what I notice afterwards is that the folder "subfolder1" that I previously moved away is still in my local onedrive.  Shouldn't it be removed by the above synchronization ?

my sync_list file in .config/onedrive is empty, and I don't skip_dir anything outside the Documents folder in my config file.

Best,
Esben


On Wed, 2023-02-08 at 10:58 -0800, abraunegg wrote:
@esbenph
Firstly, what application version are you using? What is the output of onedrive --version ?
But today I discovered that after I have moved (i.e. archived) sub-folders from my Documents folder to other places on my OneDrive these changes are not executed in my local onedrive folders on Ubuntu.
Where did you 'move' them to? Another folder on your main 'account' or to a different SharePoint site? Is the place you 'moved' the files to 'excluded' by any client side filtering rules such as 'skip_dir' or 'sync_list' ?

**Comment by abraunegg** [comment]
@esbenph 
> onedrive --enable-logging --log-dir /home/esben/onedrivelog/ --single-directory "Documents" --synchronize

So this is the reason *why* the move was not made. 

Because of the command you used, you are syncing only the 'Documents' folder - not all your folders under '~/Onedrive/' ... so the move you made online would have actually been discarded because of the command you used to limit the sync to just your 'Documents' folder.

Had you been syncing your entire tree using `onedrive --enable-logging --log-dir /home/esben/onedrivelog/ --synchronize` the move online would have reflected locally.

**Comment by esbenph** [comment]
No! You must misunderstand what I wrote.

I am only interested in synchronizing the Documents (and subfolders hereof). I don't understand why this move locally does not manifest as changes in the Documents folders?? Because the move of subfolder1 from Documents to another place must imply that subfolder1 is removed from Documents. Right?

I don't want to synchronize anything outside the Documents folder. Because such files I only want to access via a browser.

I my method to synchronize the Documents folder (and nothing outside this) is not the correct one. How should I then do it?

Best,
Esben


On Wed, 2023-02-08 at 11:51 -0800, abraunegg wrote:
@esbenph
onedrive --enable-logging --log-dir /home/esben/onedrivelog/ --single-directory "Documents" --synchronize
So this is the reason why the move was not made.
Because of the command you used, you are syncing only the 'Documents' folder - not all your folders under '~/Onedrive/' ... so the move you made online would have actually been discarded because of the command you used to limit the sync to just your 'Documents' folder.
Had you been syncing your entire tree using onedrive --enable-logging --log-dir /home/esben/onedrivelog/ --synchronize the move online would have reflected locally.

**Comment by abraunegg** [comment]
@esbenph 
> No! You must misunderstand what I wrote.

I have re-read and I do not think so.

> /'main-account'/Documents/subfolder1 was moved to
>/'main-account'/Archive-folder/

You moved your files from inside 'Documents' -> 'Archive-folder'

The sync command you used, explicitly syncs only the 'Documents' folder and 100% is excluding your target location. As you moved files / folders outside of that sync scope so the move to your 'Archive' folder was not replicated locally - because you are excluding that from your sync scope because of the command you used.

> And what I notice afterwards is that the folder "subfolder1" that I previously moved away is still in my local onedrive.  Shouldn't it be removed by the above synchronization ?

No it will not be removed locally because of your sync scope limitation. The local data is 'preserved' to prevent accidental data loss.

This sounds like a similar case to https://github.com/abraunegg/onedrive/issues/2292 - please have a read of that issue.

> I my method to synchronize the Documents folder (and nothing outside this) is not the correct one. How should I then do it?

This is my suggestion:

1. Configure 'sync_list' to only sync the folders you want to sync within your `~/OneDrive` folder

This way, you are keeping the sync scope as your 'whole one drive' but forcing a specific scope so that moves | adds | changes can be correctly replicated.

**Comment by esbenph** [comment]
Ok. Thank you very much for this detailed answer. Although I don’t completely understand the logic :-).  I really thought that consistently using the --single-directory feature would restrict any synchronization to this single directory, such that anything outside this area would  not be affected.
One question though. If I instead just deleted the subfolder1 from the Documents folder in my browser then I guess subfolder1 would also be deleted from my local onedrive/Documents after I synchronize. Isn’t that correct?

So what I want to achieve, I guess, is simply to upload the subfolder1 from my local onedrive/Documents to the Archive-folder via the browser (for example by dragging the folder to the appropriate browser page). And then afterwards deleting the subfolder1 from the Documents folder.

Best,
Esben

**Comment by abraunegg** [comment]
@esbenph 
> I really thought that consistently using the --single-directory feature would restrict any synchronization to this single directory, such that anything outside this area would  not be affected.

Correct with your understanding

> One question though. If I instead just deleted the subfolder1 from the Documents folder in my browser then I guess subfolder1 would also be deleted from my local onedrive/Documents after I synchronize. Isn’t that correct?

That is what should happen

> So what I want to achieve, I guess, is simply to upload the subfolder1 from my local onedrive/Documents to the Archive-folder via the browser (for example by dragging the folder to the appropriate browser page). And then afterwards deleting the subfolder1 from the Documents folder.

You need to manually delete the local data due to how you are using the application

**Comment by esbenph** [comment]
Ok. Thank you for enlightening me on how it works. I do appreciate that.

Best,
Esben
________________________________
